### PR TITLE
[pose-detection] Add WebGPU backend

### PR DESCRIPTION
**Pull request by haoyunfeix**
1. Add tfjs-webgpu backend
2. Update tfjs to 4.0.0
3. Fix backend changing issue in upload_video demo
4. Add URL backend parameter

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/1044)

### CODE REVIEW

**Comment by haoyunfeix** [comment]
@mattsoulanille @lina128 PTAL, Thanks.

**Review by mattsoulanille — CHANGES_REQUESTED** [review]
This looks good (working on Chrome Dev 108.0.5359.19 Linux), but I have a concern about surfacing errors to the user.

WebGPU is not supported on all devices, and right now, if the demo tries to switch to the WebGPU backend on an unsupported device, it freezes with no error message visible to the user. This is because the WebGPU backend [only registers itself it WebGPU is supported](https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgpu/src/base.ts#L25-L26), and in `util.js`, we [throw an error if a backend is not in the registry](https://github.com/tensorflow/tfjs-models/blob/master/pose-detection/demos/live_video/src/util.js#L39-L41).

Can we catch this error in [`index.js`](https://github.com/tensorflow/tfjs-models/blob/master/pose-detection/demos/live_video/src/index.js#L99-L101) and `alert()` the user that the backend is not supported? Ideally, we'd also switch back to the previous backend (both the tfjs runtime and the UI), but if this is difficult, then alerting the user should be sufficient.

I've discovered this issue on the live demo, but I think the upload video demo has it as well.

File: pose-detection/demos/upload_video/src/index.js [hunk 1]
```diff
@@ -212,18 +214,16 @@ async function app() {
...
   await setBackendAndEnvFlags(STATE.flags, STATE.backend);
+  detector = await createDetector();
 
   const runButton = document.getElementById('submit');
   runButton.onclick = run;
 
   const uploadButton = document.getElementById('videofile');
   uploadButton.onchange = updateVideo;
-
-  checkUpdate();
```
[the remark this answers is not in the document]
Comment (haoyunfeix) [reply pose-detection/demos/upload_video/src/index.js:226]: Thanks, removed.

File: pose-detection/demos/upload_video/src/option_panel.js [hunk 2]
```diff
@@ -92,8 +94,10 @@ async function showBackendConfigs(folderController) {
             .__controllers[folderController.__controllers.length - 1]);
   }
   const backends = params.MODEL_BACKEND_MAP[params.STATE.model];
-  // The first element of the array is the default backend for the model.
-  params.STATE.backend = backends[0];
+  if(params.STATE.backend === undefined || params.STATE.backend === null) {
```
[the remark this answers is not in the document]
Comment (haoyunfeix) [reply pose-detection/demos/upload_video/src/option_panel.js]: Thanks, that doc is helpful!

File: pose-detection/demos/live_video/src/util.js [hunk 3]
```diff
@@ -37,7 +38,14 @@ export function isMobile() {
 async function resetBackend(backendName) {
   const ENGINE = tf.engine();
   if (!(backendName in ENGINE.registryFactory)) {
-    throw new Error(`${backendName} backend is not registed.`);
+    if(backendName === 'webgpu') {
+      alert(`${backendName} backend is not registered. Maybe should use a WebGPU enabled target e.g. chrome canary with --enable-unsafe-webgpu flag`);
```
Comment (mattsoulanille) [inline pose-detection/demos/live_video/src/util.js:42]:
```suggestion
      alert('webgpu backend is not registered. Your browser may not support WebGPU yet. To test this backend, please use a supported browser, e.g. Chrome canary with --enable-unsafe-webgpu flag');
```

**Review by mattsoulanille — APPROVED** [review]
Thanks for adding the error message when WebGPU is not supported.

**Review by lina128 — APPROVED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/1044#-:-NFtZgah6Yi2mpXGBbzx:bb6ywmx)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @haoyunfeix and @mattsoulanille)

**Pull request merged**